zbuf/reader.go

```diff
@@ -0,0 +1,80 @@
+package zbuf
+
+import (
+	"github.com/brimdata/zed"
+	"github.com/brimdata/zed/zio"
+)
+
+type Control struct {
+	Message interface{}
+}
+
+var _ error = (*Control)(nil)
+
+func (c *Control) Error() string {
+	return "control"
+}
+
+type SetChannel int
+type EndChannel int
+
+type noControl struct {
+	zio.Reader
+}
+
+func NoControl(r zio.Reader) *noControl {
+	return &noControl{Reader: r}
+}
+
+func (n *noControl) Read() (*zed.Value, error) {
+	for {
+		val, err := n.Reader.Read()
+		if err != nil {
+			if _, ok := err.(*Control); ok {
+				continue
+			}
+		}
```

Comment:
Type assertion will fail if `err` is nil.
```suggestion
		if _, ok := err.(*Control); ok {
			continue
		}
```